core-bundle/src/Twig/FragmentTemplate.php

```diff
@@ -0,0 +1,719 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Twig;
+
+use Contao\CoreBundle\Asset\ContaoContext;
+use Contao\FilesModel;
+use Contao\Model;
+use Contao\Template;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * This class is a simple container object for template data.
+ */
+final class FragmentTemplate /* TODO: remove base class in Contao 6 */ extends Template
+{
+    /**
+     * @var array<string,mixed>
+     */
+    private array $context = [];
+
+    /**
+     * @param \Closure(self, Response|null):Response $onGetResponse
+     *
+     * @internal
+     * @noinspection MagicMethodsValidityInspection
+     */
+    public function __construct(private string $templateName, private \Closure $onGetResponse)
+    {
+        // Do not call parent constructor
+    }
+
+    /**
+     * @param string $key
+     * @param mixed  $value
+     */
+    public function __set(/* string */ $key, $value): void
+    {
+        $this->set($key, $value);
+    }
+
+    /**
+     * @param string $key
+     *
+     * @return mixed
+     */
+    public function __get(/* string */ $key)
+    {
+        return $this->get($key);
+    }
+
+    /**
+     * @param string $key
+     */
+    public function __isset(/* string */ $key): bool
```

Comment:
Why don‘t we adjust the parent class instead?

Reply:
We could, but as the base class is `Template`, I'm not sure if this will unnecessarily break things in the legacy framework. Shall we? :grin: 

Btw., there are also new and typed methods `set` `get` and `has`, that I would prefer using anyways.

Reply:
Yes. We are allowed to break BC in Contao 5.0.

Reply:
see d0d73f1f8b37062f39d00acf6a64b95c6702ff82

Reply:
I like. 👍